.github/workflows/update_dependencies.yml

```diff
@@ -0,0 +1,80 @@
+on:
+  schedule:
+    # Run every 10th of the month, as it seems golang.org/x/... packages are usually
+    # released at the beginning of the month.
+    # Run at 6AM UTC so that it's ready at the beginning of the day CET.
+    - cron: "0 6 10 * *"
+
+  # allow triggering the workflow manually
+  workflow_dispatch:
+
+permissions: {}
+
+jobs:
+  backport:
+    name: Update golang.org/x/... dependencies
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # push commit and branch
+      pull-requests: write # create a PR
+    steps:
+      - uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+        with:
+          # credentials are needed to create the PR at the end of the workflow
+          persist-credentials: true
+      - uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
+        with:
+          go-version-file: ".go-version"
+      - name: Update every golang.org/x/... package
+        run: |
+          go get -u golang.org/x/arch golang.org/x/crypto \
+                    golang.org/x/lint golang.org/x/mod \
+                    golang.org/x/net golang.org/x/sync \
+                    golang.org/x/sys golang.org/x/term \
+                    golang.org/x/text golang.org/x/time \
+                    golang.org/x/xerrors golang.org/x/tools \
+                    golang.org/x/exp
+          # run tidy twice because the first one doesn't always clean everything
+          # depending on module order
+          inv -e tidy tidy
+      - uses: stefanzweifel/git-auto-commit-action@8621497c8c39c72f3e2a999a26b4ca1b5058a842 # v5.0.1
+        id: autocommit
+        env:
+          BRANCH_NAME: update-golang-org-x-${{ github.run_id }}-${{ github.run_attempt }}
+        with:
+          commit_message: "chore(deps): update all golang.org/x/... dependencies"
+          branch: $BRANCH_NAME
+          create_branch: true
+          skip_fetch: true
+      - uses: actions/create-github-app-token@5d869da34e18e7287c1daad50e0b8ea0f506ce69 # v1.11.0
+        id: app-token
+        with:
+          app-id: ${{ vars.DD_GITHUB_TOKEN_GENERATOR_APP_ID }}
+          private-key: ${{ secrets.DD_GITHUB_TOKEN_GENERATOR_PRIVATE_KEY }}
+      - name: Create PR
+        env:
+          TMP_PR_BODY_PATH: /tmp/pr_body
+          github_token: ${{ steps.app-token.outputs.token }}
+          PR_TITLE: "[automated] Update golang.org/x/... dependencies"
+          PR_LABELS: "team/agent-shared-components,qa/done,changelog/no-changelog"
+          GITHUB_REF: ${{ github.ref }}
+        run: |
+          # Generate the PR description
+          cat > $TMP_PR_BODY_PATH <<- EOM
+          ### What does this PR do?
+          Update all `golang.org/x/...` dependencies.
+
+          ### Motivation
+          Keep dependencies up-to-date, benefit from bug fixes and new features.
+
+          ### Describe how you validated your changes
+          CI is considered enough to validate changes.
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Does it create a PR with empty spaces at the start of each line?

Reply:
I don't think so but I'm about to test it so I'll make sure

Reply:
Yep I made sure it was correct, and fixed a couple other issues 😄